### PR TITLE
Use tool.scikit-build.cmake.version, set scikit-build-core minimum-version

### PR DESCRIPTION
**Pull request by jameslamb**
Contributes to https://github.com/rapidsai/build-planning/issues/58.

`scikit-build-core==0.10.0` was released today (https://github.com/scikit-build/scikit-build-core/releases/tag/v0.10.0), and wheel-building configurations across RAPIDS are incompatible with it.

This proposes upgrading to that version and fixing configuration here in a way that:

* is compatible with that new `scikit-build-core` version
* takes advantage of the forward-compatibility mechanism (`minimum-version`) that `scikit-build-core` provides, to reduce the risk of needing to do this again in the future

### CODE REVIEW

File: conda/environments/all_cuda-118_arch-x86_64.yaml
Comment (jameslamb) [inline conda/environments/all_cuda-118_arch-x86_64.yaml:28]: These were pulled in by the updates from https://github.com/rapidsai/rapids-metadata/pull/24.

This is a good change that should get onto 24.08 even if this PR ends up getting re-targeted to 24.10.

**Comment by jameslamb** [comment]
CI is in a bad state here, I don't *think* it's a result of this PR's changes.

We'll need to merge https://github.com/rapidsai/devcontainers/pull/377 to fix the devcontainers builds. And then these others to fix other parts of CI:

* #1428
* #1429

I'll check back on this after merging those, before putting this up for review.

**Comment by jameslamb** [comment]
Reported the test failures in #1433.

The changes in this PR are ready for review.

**Comment by jakirkham** [comment]
Restarting failed CI jobs now that `devcontainers` have been updated

**Review by jakirkham — APPROVED** [review]

**Comment by jameslamb** [comment]
This will be blocked until #1434 is complete

**Comment by vyasr** [comment]
Restarted again now that #1434 is merged.

**Pull request merged**
